core/state/state_object.go

```diff
@@ -242,6 +252,7 @@ func (s *stateObject) SetState(key, value common.Hash) {
 		prevalue: prev,
 	})
 	s.setState(key, value)
+	s.isFilled[key] = false
 }
 
```

Comment:
General comment, and might be part of WIP: sounds like this `isFilled` map should be journal-compatible to handle reverts/reorgs/persistence?